consensus/core/src/transactions_client.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::block::Transaction;
+use crate::context::Context;
+use mysten_metrics::metered_channel;
+use mysten_metrics::metered_channel::channel_with_total;
+use std::sync::Arc;
+use tap::tap::TapFallible;
+use thiserror::Error;
+use tracing::error;
+
+/// Maximum number of transactions to be fetched per request of `next`
+const MAX_FETCHED_TRANSACTIONS: usize = 100;
+/// The maximum number of transactions pending to the queue to be pulled for block proposal
+const MAX_PENDING_TRANSACTIONS: usize = 2_000;
+
+/// The TransactionsConsumer is responsible for fetching the next transactions to be included for the block proposals.
+/// The transactions are submitted to a channel which is shared between the TransactionsConsumer and the TransactionsClient
+/// and are pulled every time the `next` method is called.
+#[allow(dead_code)]
+pub(crate) struct TransactionsConsumer {
+    tx_receiver: metered_channel::Receiver<Transaction>,
+    max_fetched_per_request: usize,
+}
+
+#[allow(dead_code)]
+impl TransactionsConsumer {
+    pub(crate) fn new(tx_receiver: metered_channel::Receiver<Transaction>) -> Self {
+        Self {
+            tx_receiver,
+            max_fetched_per_request: MAX_FETCHED_TRANSACTIONS,
+        }
+    }
+
+    pub(crate) fn with_max_fetched_per_request(mut self, max_fetched_per_request: usize) -> Self {
```

Comment:
Why not just set this in `new()`?

Reply:
basically I wanted to avoid start clattering the `new` with optional parameters , that's why I considered just using an additional method. I guess we could do it in `new` too but in common case it won't be needed much